### PR TITLE
Make call function generic.

### PR DESCRIPTION
**Pull request by MeirShpilraien**

### CODE REVIEW

File: src/redismodule.rs [hunk 1]
```diff
@@ -108,6 +114,14 @@ impl RedisString {
         Self { ctx, inner }
     }
 
+    pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[u8]) -> Self {
```
Comment (gkorland) [inline src/redismodule.rs:117]:
```suggestion
    #[allow(clippy::not_unsafe_ptr_arg_deref)]
    pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[u8]) -> Self {
```

Comment (gkorland) [inline src/redismodule.rs:117]:
```suggestion
    #[allow(clippy::not_unsafe_ptr_arg_deref)]
    pub fn create_from_slice(ctx: *mut raw::RedisModuleCtx, s: &[u8]) -> Self {
```

File: src/redismodule.rs [hunk 2]
```diff
@@ -108,6 +114,14 @@ impl RedisString {
...
+        let inner = unsafe {
+            raw::RedisModule_CreateString.unwrap()(ctx, s.as_ptr() as *const c_char, s.len())
```
Comment (gkorland) [inline src/redismodule.rs:119]:
```suggestion
            raw::RedisModule_CreateString.unwrap()(ctx, s.as_ptr().cast::<c_char>(), s.len())
```

File: src/context/mod.rs [hunk 3]
```diff
@@ -28,6 +28,60 @@ pub struct Context {
     pub ctx: *mut raw::RedisModuleCtx,
 }
 
+pub struct StrCallArgs<'a> {
+    is_owner: bool,
+    args: Vec<*mut raw::RedisModuleString>,
+    phantom: std::marker::PhantomData<&'a u64>, // allows to make sure the object will not leave longer than actaull arguments slice
```
Comment (gkorland) [inline src/context/mod.rs:34]:
```suggestion
    // Phantom is used to make sure the object will not live longer than actual arguments slice
    phantom: std::marker::PhantomData<&'a raw::RedisModuleString>,
```

File: src/context/mod.rs [hunk 4]
```diff
@@ -28,6 +28,60 @@ pub struct Context {
...
+}
+
+impl<'a> Drop for StrCallArgs<'a> {
+    fn drop(&mut self) {
+        if self.is_owner {
+            for v in self.args.iter() {
+                unsafe { raw::RedisModule_FreeString.unwrap()(std::ptr::null_mut(), *v) };
+            }
```
Comment (gkorland) [inline src/context/mod.rs:42]:
```suggestion
            self.args.iter_mut().for_each(|v| unsafe {
                raw::RedisModule_FreeString.unwrap()(std::ptr::null_mut(), *v)
            });
```

**Review by gkorland — CHANGES_REQUESTED** [review]

**Comment by gkorland** [comment]
@MeirShpilraien few minor comments

**Comment by MeirShpilraien** [comment]
@gkorland merged

**Comment by gkorland** [comment]
@MeirShpilraien some conflicts

**Comment by MeirShpilraien** [comment]
@gkorland fixed.

**Review by gkorland — APPROVED** [review]

**Pull request merged**